app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -21,7 +21,7 @@ def create
       result = otp_verification_form.submit
       post_analytics(result)
       if result.success?
-        handle_valid_otp
+        handle_valid_otp(nil, auth_method: params[:otp_delivery_preference])
```

Comment:
If possible I'd convert that first argument to a keyword as well, so we don't have to wonder what the `nil` means

Reply:
Good call, changed in b40aa8e12b

Reply:
big fan, thank you!